content/blog/2024/kubewarden-1-10-release.md

```diff
@@ -0,0 +1,128 @@
+---
+title: Kubewarden 1.10 release
+authors:
+  - Víctor Cuadrado
+date: 2024-01-24
+---
+
+We have the first release of 2024, Kubewarden 1.10.0! 🎉🥳
+And this one contains a nice bag of goodies, let's have a look!
+
+## Policy server has reduced memory usage and increased reliability
+
+A nice graph is worth a thousand words!
+
+{{<figure src="/images/policy-server-optimization-1.10.png" alt="graph of the policy-server memory optimization"  >}}
+
+Note the slightly lower memory consumption, and unchanging consumption when
+scaling horizontally.
+
+This graph represents the memory consumption of one instance of policy-server,
+containing 13 policies:
+
+- 4 instances of "verify-signatures"
+- 5 of "pod-privileged"
+- 2 of "go-wasi-template" (a 20MB policy, WASI being experimental)
+- 1 Rego policy
+- 1 normal Rust policy
+
+The policy-server was configured with one worker to start, progressing to eight. For each worker
+value, 10 samplings were taken.
+
+#### Constant memory consumption when scaling horizontally
+
+Before, we needed one instance of the policy evaluator (hence the Wasmstime stack)
+for each policy deployed.
+
+Now, memory consumption is constant no matter how many parallel policy
+evaluators are spawned when scaling horizontally, by increasing the number of
+threads available in the machine. This is achieved by only creating an instance
+of the policy evaluator per unique type of policy module.
+
+For example, if a user has the apparmor policy deployed 5 times (with different
+names, settings, etc), now only one instance of PolicyEvaluator will be
+allocated for all of those 5 deployments.
+
+In addition, we rely now on Wasmtime's pre-initialization of Wasm modules. This
+allows us to keep a deduplicated list of all the policies in use, in their
+pre-initialized form. This list is shared between all policy evaluators in the
+policy-server.
+
+#### Protection against policy memory leaks, and clean slate per policy
+
+Before, there were long-running policy evaluators, ready to process policy requests.
+
+Now, policy evaluation mode happens on-demand: we receive a request, a fresh
+policy evaluator is created, used just for evaluation that request, and
+discarded.
+
+This is possible as we now leverage Wasmtime's pre-initialization, hence
+creation time of policy evaluators in greatly reduced.
+
+Using this on-demand approach to policy evaluators brings two main advantages:
+
+- Impact of policies leaking memory is now virtualy non-existent.
+- Each evaluation starts with a clean slate, preventing bugs caused by policies
+  leaving unclean state between evaluations.
+
+#### Codebase simplification and increased test coverage
```

Comment:
@jordojordo  here, the css is missing what seems to be a bottom margin for the `ul`:

![image](https://github.com/kubewarden/kubewarden.io/assets/2196685/071a56c1-4913-49d6-8409-8154646f7236)

Since this comes from the Hugo theme and I'm not sure how to tackle it, could you have a look? It may be just some secs for you.

Reply:
@viccuad That seems to be coming from [this css rule](https://github.com/rancherlabs/projects-theme/blob/5e3258f90352c9558674a2cbf5148e8bc8984893/assets/sass/main.scss#L82) - It would be possible to override it, would you like me to override it for all of the `ul` elements within the blog?

Reply:
Thanks! I think it would be a good idea, yes.

Reply:
I've added `10px` to the bottom margin for blog `ul` to match the top margin:

![margin](https://github.com/kubewarden/kubewarden.io/assets/40806497/2c3b5430-8a01-4bb5-8f21-964024a85c1e)